LambdaEngine/Include/Memory/API/StackAllocator.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+#include "LambdaEngine.h"
+
+namespace LambdaEngine
+{
+	/*
+	* MemoryArena
+	*/
+
+	class MemoryArena
+	{
+	public:
+		inline MemoryArena(uint32 sizeInBytes = 4096)
+			: m_pMemory(nullptr)
+			, m_Offset(0)
+			, m_SizeInBytes(sizeInBytes)
+		{
+			m_pMemory = Malloc::AllocateType<byte>(m_SizeInBytes);
+			ZERO_MEMORY(m_pMemory, m_SizeInBytes);
```

Comment:
Is there a particular reason you zero the memory? I noticed `MemoryArena::Pop` doesn't zero freed memory.

Reply:
I used this as a debug thing, probably does not need it in production.